Fix rendering of group members

ReeceHoffmann opened the pull request:
Changes:
  - converts groups view to use react-query
  - removes older data handling and networking boilerplate made redundant by the change to react-query
  - updates and removes tests as needed to handle new data fetching logic

ReeceHoffmann committed:
temp commit revert later

ReeceHoffmann committed:
fix: restore group member rendering

ReeceHoffmann committed:
Merge branch 'main' into fix-rendering-of-group-members

ReeceHoffmann committed:
chore: remove unused boilerplate

ReeceHoffmann committed:
chore: address deepsource concerns

igboyes commented on src/js/app/websocket.ts:
```diff
@@ -16,12 +16,14 @@ import { QueryClient } from "react-query";
     roles: roleKeys,
     users: userKeys,
     uploads: fileKeys,
+    groups: groupKeys,
```
Alphabetize.

igboyes commented on src/js/groups/__tests__/actions.test.js:
```diff
@@ -1,29 +1,13 @@
         expect(result).toEqual({
             type: WS_INSERT_GROUP,
-            payload: { ...data }
+            payload: { ...data },
```
?
```suggestion
            payload: data,
```

ReeceHoffmann replied:
Worth it. (jk fixed)

igboyes commented on src/js/groups/querys.ts:
```diff
@@ -0,0 +1,95 @@
+import { Request } from "../app/request";
+import { groupNameUpdate } from "./types";
+
+export const groupKeys = {
```
```suggestion
export const groupQueryKeys = {
```

igboyes commented on src/js/groups/querys.ts:
```diff
@@ -0,0 +1,95 @@
+
+function listGroups() {
+    return Request.get("/groups").then(response => response.body);
+}
```
I think the API functions should be in a separate module.

igboyes commented on src/js/groups/types.ts:
```diff
@@ -1 +1,24 @@
+export type groupNameUpdate = {
+    id: string;
+    name?: string;
+};
```
I think this is convention.

```suggestion
export type GroupNameUpdate = {
    id: string;
    name?: string;
};
```

igboyes commented on src/tests/fake/groups.ts:
```diff
@@ -14,3 +17,27 @@ export function createFakeGroupMinimal(props?: createFakeGroupMinimalProps): Gro
+    };
+}
+
+export function mockGetGroupAPI(group: Group) {
```
I think it might be faster to grok if you standardized on `mockApiGetGroup`. Then we know it is for API mocking from the start instead of having to read to the end of the variable name.

```suggestion
export function mockGetGroupApi(group: Group) {
```

igboyes commented on src/tests/fake/groups.ts:
```diff
@@ -14,3 +17,27 @@ export function createFakeGroupMinimal(props?: createFakeGroupMinimalProps): Gro
+    return nock("http://localhost").get(`/api/groups/${group.id}`).reply(200, group);
+}
+
+export function mockListGroupsAPI(groups: Group[]) {
```
```suggestion
export function mockApiListGroups(groups: Group[]) {
```

igboyes commented on src/tests/fake/groups.ts:
```diff
@@ -14,3 +17,27 @@ export function createFakeGroupMinimal(props?: createFakeGroupMinimalProps): Gro
     return merge(defaultGroupMinimal, props);
 }
+
+type createFakeGroupProps = {
```
```suggestion
type CreateFakeGroupProps = {
```

igboyes reviewed (changes requested):
At a minimum, I want to start documenting functions and variables that are widely reused. You have quite a few in this PR...

ReeceHoffmann committed:
chore: address review

  - add jsdocs for groups query methods
  - add jsdocs for files, permissions, users, and groups faking functions

ghost commented:
:tada: This PR is included in version 3.12.3 :tada:

The release is available on [GitHub release](https://github.com/virtool/virtool-ui/releases/tag/3.12.3)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
